client/src/document/forms/manage.tsx

```diff
@@ -0,0 +1,99 @@
+import React from "react";
+import { useDocumentURL } from "../hooks";
+import { useParams } from "react-router";
+import { Link } from "react-router-dom";
+import useSWR from "swr";
+
+import "./manage.scss";
+
+function DocumentDelete({ slug, locale }) {
+  async function deleteDocument() {
+    if (
+      !window.confirm(
+        "Are you sure you want to delete this document and all its children?"
+      )
+    ) {
+      return;
+    }
+    const response = await fetch(
+      `/_document?slug=${encodeURIComponent(slug || "")}&locale=${locale}`,
+      {
+        method: "DELETE",
+        headers: { "Content-Type": "application/json" },
+      }
+    );
+    if (response.ok) {
+      window.alert("Document successfully deleted");
+    } else {
+      window.alert(`Error while deleting document: ${response.statusText}`);
+    }
+  }
+
+  return (
+    <div>
+      <button onClick={deleteDocument}>delete</button>
```

Comment:
Specify `type` attribute:

```suggestion
      <button type="button" onClick={deleteDocument}>delete</button>
```